raft/src/main/java/org/apache/kafka/raft/BatchReader.java

```diff
@@ -57,4 +60,9 @@
      */
     @Override
     void close();
+
+    static <T> BatchReader<T> singleton(Batch<T> batch) {
```

Comment:
Can you add JavaDoc for this new method?

Reply:
I removed this method. I think it is a dangerous method to have in `BatchReader` since batch readers created this way do not integrate with the rest of the `KafkaRaftClient`. This method was only used for tests so I just modified those cases to use `MemoryBatchReader` directly.